test/CMakeLists.txt

```diff
@@ -13,28 +13,30 @@ function(create_test)
     ${ARGN}
     )
 
-  add_executable(${ARG_TARGET}
-    ${ARG_SOURCES}
-    main.cpp
-    )
+  add_executable(${ARG_TARGET} ${ARG_SOURCES} MainMPI.cpp)
   target_link_libraries(${ARG_TARGET}
-    PRIVATE Catch2::Catch2 ${PROJECT_NAME}::${LIB_NAME}
+    PRIVATE doctest::doctest ${PROJECT_NAME}::${LIB_NAME}
     )
   target_include_directories(${ARG_TARGET}
-    PRIVATE ${CMAKE_SOURCE_DIR}
+    PRIVATE ${CMAKE_SOURCE_DIR} ${DOCTEST_INCLUDE_DIR}
     )
-  foreach(numProcs RANGE 1 ${ARG_MAX_NUM_PROCS})
-    add_test(NAME ${ARG_TARGET}_n${numProcs} COMMAND ${MPIEXEC} --oversubscribe ${MPIEXEC_NUMPROC_FLAG} ${numProcs} ${MPIEXEC_PREFLAGS} ./${ARG_TARGET})
-  endforeach()
```

Comment:
what's the reason for removing this part?

I have just realised we could use something like this in nextsimdg (if it works with `doctest`)

Reply:
Ah I'd been removing `MAX_NUM_PROCS` stuff but this part would potentially be useful. Will try adding back in.